src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConditionalOperatorTests.cs

```diff
@@ -2326,5 +2326,88 @@ static IEnumerable<int> Where<TSource>()
 
             CompileAndVerify(source);
         }
+
+        [Fact, WorkItem(17756, "https://github.com/dotnet/roslyn/issues/17756")]
+        public void TestConditionalOperatorNotLvalue()
+        {
+            var source = @"
+    class Program
+    {
+
+        interface IIncrementable
+        {
+            int Increment();
+        }
+
+        struct S1: IIncrementable
+        {
+            public int field;
+            public int Increment() => field++;
+        }
+
+        static void Main()
+        {
+            S1 v = default(S1);
+            v.Increment();
+
+            (true ? v : default(S1)).Increment();
+
+            System.Console.WriteLine(v.field);
+
+            System.Console.WriteLine((true ? v : default(S1)).Increment());
+
+            System.Console.WriteLine(v.field);
+
+            Test(ref v);
+            System.Console.WriteLine(v.field);
+        }
+
+        static void Test<T>(ref T arg) where T:IIncrementable
+        {
+            (true ? arg : default(T)).Increment();
+        }
+    }
+";
+            string expectedOutput = @"1
+1
+1
+1";
+            CompileAndVerify(source, expectedOutput: expectedOutput);
+        }
```

Comment:
Not sure if this is strictly necessary since it is completely symmetrical, but it is easy to change one scenario to use "false"

---
In reply to: [108731184](https://github.com/dotnet/roslyn/pull/18166#discussion_r108731184) [](ancestors = 108731184)